src/gui/qgsmapcanvasmap.h

```diff
@@ -16,22 +16,18 @@
 #ifndef QGSMAPCANVASMAP_H
 #define QGSMAPCANVASMAP_H
 
-#include <QGraphicsRectItem>
-#include <QPixmap>
-#include <QTimer>
-
-#include <qgis.h>
-#include <qgsmaptopixel.h>
-
 #include <qgsmapcanvasitem.h>
 
 class QgsMapSettings;
 class QgsMapCanvas;
 
 /** \ingroup gui
  * A rectangular graphics item representing the map on the canvas.
+ *
+ * @note This class is not a part of public API
+ * @note not available in Python bindings
```

Comment:
Probably should wrap this whole bit (and the cpp implementations) in a doxygen

/// @cond PRIVATE

/// @endcond

That will hide it from the generated docs too...